Improve cmake external project compatibility

Shelnutt2 opened the pull request:
This converts all `CMAKE_SOURCE_DIR` to `CMAKE_CURRENT_SOURCE_DIR` and  `CMAKE_BINARY_DIR` to `CMAKE_CURRENT_BINARY_DIR`. This ensures the project builds correctly regardless if it is the top level cmake project or not.

If you'd like these two changes squashed into a single commit, as in #602 let me know.

Closes #597

tdenniston commented on CMakeLists.txt:
```diff
@@ -31,9 +31,9 @@
 
-set(TILEDB_CMAKE_INPUTS_DIR "${CMAKE_SOURCE_DIR}/cmake/inputs")
+set(TILEDB_CMAKE_INPUTS_DIR "${CMAKE_CURRENT_SOURCE_DIR}/cmake/inputs")
 
```
It might be nice here to define a `TILEDB_ROOT_SOURCE_DIR` variable which can then be used in subdirectories instead of having to do `${CMAKE_CURRENT_SOURCE_DIR}/..` in several places.

tdenniston replied:
After some discussion with @jakebolewski it's fine to leave this as-is (don't worry about adding the new variable). He also made the point that in navigating nested CMake projects, the relative path may actually be clearer (rather than having to look up the definition of a project-specific variable).

Shelnutt2 replied:
I've left this as `CMAKE_CURRENT_SOURCE_DIR` and not moved to `TILEDB_ROOT_SOURCE_DIR`. I agree with @jakebolewski that using `CMAKE_CURRENT_SOURCE_DIR` makes it easy to know the path based on the `CMakeLists.txt` file that is being examined.

tdenniston commented:
I left one comment, otherwise I like it. Go ahead and squash the two commits (and rebase onto `dev` again) -- in general we try to keep PRs to a single commit unless there is sufficient complexity in the PR to benefit from multiple commits.

I am not sure why the macOS Travis build stalled, let's see if it happens again.

Shelnutt2 commented:
@tdenniston Thank you for the guidance on using single commits for these relatively small changes.

I've squashed and rebased. Looks like Travis CI OSX builds are having problems with [delayed builds now](https://www.traviscistatus.com/incidents/ltw1fkpz5m3t). Once the OSX build runs and passes this should be good to merge unless there are other changes needed.

tdenniston commented:
@Shelnutt2 it looks like you may still need to rebase the commit on top of `dev`.

Shelnutt2 commented:
@tdenniston rebased again now that the GNUInstallDirs branch was merged.

tdenniston commented:
Ok, great. Thanks for your patience, hopefully they'll sort out their macOS issues soon.
